raft/src/main/java/org/apache/kafka/raft/QuorumState.java

```diff
@@ -663,11 +663,12 @@ public boolean isUnattached() {
     }
 
     public boolean isUnattachedNotVoted() {
-        return isUnattached() && !unattachedStateOrThrow().votedKey().isPresent();
+        EpochState fixedState = state;
+        return fixedState instanceof UnattachedState && !((UnattachedState) fixedState).votedKey().isPresent();
     }
 
     public boolean isUnattachedAndVoted() {
-        return isUnattached() && unattachedStateOrThrow().votedKey().isPresent();
+        return maybeUnattachedState().map(UnattachedState::votedKey).isPresent();
```

Comment:
This doesn't look correct. I think `maybeUnattachedState().map(UnattachedState::votedKey)` has the type `Optional<Optional<ReplicaKey>>`.